protocol/synthetix/contracts/storage/RewardDistribution.sol

```diff
@@ -112,6 +112,63 @@ library RewardDistribution {
         }
     }
 
+    /**
+     * @dev Gets the total shares of a reward distribution entry.
+     */
+    function getEntry(
+        Data storage self,
+        uint256 totalSharesAmountD18
+    ) internal view returns (int256) {
+        // No balance if a pool is empty or if it has no rewards
+        if (self.scheduledValueD18 == 0 || totalSharesAmountD18 == 0) {
+            return 0;
+        }
+
+        uint256 curTime = block.timestamp;
+
+        int256 valuePerShareChangeD18 = 0;
+
+        // No balance if current time is before start time
+        if (curTime < self.start) {
+            return 0;
+        }
+
+        // If the entry's duration is zero and the its last update is zero,
+        // consider the entry to be an instant distribution.
+        if (self.duration == 0 && self.lastUpdate < self.start) {
```

Comment:
Are there cases when duration is 0 and we have made an update? 
Can the if be just `self.duration == 0`? Otherwise we have to watch out for dividing with 0 in the else

Reply:
Not from my understanding. RewardDistribution.lastEntry in this case will be zero
https://github.com/Synthetixio/synthetix-v3/blob/main/protocol/synthetix/contracts/storage/RewardDistribution.sol#L109